### PR TITLE
Support Protobuf 5.x

### LINKED ISSUE #32302: [Feature Request]: Protobuf 5+ support
### What would you like to happen?

Is it possible for the python sdk and image to support protobuf 5+?

I have a project that consumes protobuf files generated with version 5.27 of protoc. Those files introduce the following import which isn't understood by protobuf 4.26.0.

```
from google.protobuf import runtime_version as _runtime_version
```

### Issue Priority

Priority: 2 (default / most feature requests should be filed as P2)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Infrastructure
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by jrmccluskey**
Updates dependency ranges to support protobuf 5.x. Also adds custom timestamp conversion code, as a breaking change in protobuf 5.x began enforcing a timestamp range narrower than what had previously been supported in beam.

Closes #32302

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/python/apache_beam/utils/proto_utils.py [hunk 1]
```diff
@@ -36,6 +36,10 @@
 
 message_types = (message.Message, )
 
+_SECONDS_TO_MICROS = 10**6
+_SECONDS_TO_NANOS = 10**8
```
[the remark this answers is not in the document]
Comment (jrmccluskey) [reply sdks/python/apache_beam/utils/proto_utils.py]: actually that constant is unused? looks like a holdover from ritesh's testing, I can just remove it

File: sdks/python/apache_beam/utils/proto_utils_test.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class TestProtoUtils(unittest.TestCase):
+  def test_from_micros_duration(self):
+    ts = proto_utils.from_micros(duration_pb2.Duration, MAX_TIMESTAMP.micros)
+    expected = duration_pb2.Duration(
+        seconds=MAX_TIMESTAMP.seconds(), nanos=775000000)
+    self.assertEqual(ts, expected)
+
+  def test_from_micros_timestamp(self):
```
Comment (shunping) [inline sdks/python/apache_beam/utils/proto_utils_test.py:33]: Are there any tests to cover the new function to_micros()? If not, I suggest we add that too.

Comment (jrmccluskey) [reply sdks/python/apache_beam/utils/proto_utils_test.py:33]: added extra unit tests for `to_micros()` plus some tests through both functions to make sure we were getting the same values back out.

Comment (shunping) [reply sdks/python/apache_beam/utils/proto_utils_test.py:33]: Thanks!

**Comment by Abacn** [comment]
looks like this is breaks https://github.com/apache/beam/actions/workflows/beam_PostCommit_Python_Dependency.yml

a PostCommit check several ML dependency version compatibility

error message:

```
py39-transformers-428: commands_pre[2]> pip check
tensorflow 2.12.0 has requirement protobuf!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.20.3, but you have protobuf 5.29.0rc1.
```

**Pull request merged**